jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -194,7 +204,13 @@ def _parse_esa_s3d(app, hdulist, data_label, ext='DATA', viewer_name='flux-viewe
     wcs = WCS(wcs_dict)
     flux = np.moveaxis(flux, 0, -1)
     flux = np.swapaxes(flux, 0, 1)
-    data = Spectrum1D(flux, wcs=wcs)
+
+    metadata = standardize_metadata(hdulist[ext].header)
+    metadata.update(wcs_dict)  # To be internally consistent
+    if hdulist[ext].name != 'PRIMARY' and 'PRIMARY' in hdulist:
+        metadata[PRIHDR_KEY] = standardize_metadata(hdulist['PRIMARY'].header)
```

Comment:
this seems to be a re-used block of code in a few places... could this logic be moved entirely into `standardize_metadata` or no?

Reply:
Not really, as passing in `hdulist` into `standardize_metadata` would make it too specific to FITS, which should stay in the realm of parsers.

Reply:
I mean `standardize_metadata` already checks for `fits.Header` but you could embed `fits.Header` in a plain `dict`, which is what `specutils` does, but one would/should never put the whole `HDUList` inside metadata.